### PR TITLE
Update release tooling and configuration

### PR DESCRIPTION
**Pull request by danudey**
Porting forward the release tooling fixes and updates from v3.26.

### CODE REVIEW

File: hack/release/Makefile [hunk 1]
```diff
@@ -10,6 +10,21 @@ release: .vm.created
 	./remote-execute.sh "cd calico && git fetch --tags && git checkout $(GIT_BRANCH)"
 	./remote-execute.sh "cd calico && make release"
 
+git-fetch-remote: .vm.created
+	./remote-execute.sh "cd calico && git fetch --tags && git checkout $(GIT_BRANCH)"
+
+update-versions-remote: .vm.created
+	./remote-execute.sh "cd calico && make update-versions"
+
+generate-manifests-remote:
```
Comment (matthewdupre) [inline hack/release/Makefile:19]: Should this depend on .vm.created?

Comment (matthewdupre) [reply hack/release/Makefile:19]: (Might not matter much if this isn't the first step to be ran)

**Review by matthewdupre — APPROVED** [review]

File: hack/release/Makefile (hunk 1 above)
Comment (danudey) [reply hack/release/Makefile:19]: Yes, it should. Thanks!

File: hack/release/pkg/builder/builder.go [hunk 2]
```diff
@@ -288,7 +288,7 @@ func (r *ReleaseBuilder) publishPrereqs(ver string) error {
 //
 // - release-vX.Y.Z.tgz: contains images, manifests, and binaries.
 // - tigera-operator-vX.Y.Z.tgz: contains the helm v3 chart.
-// - calico-windows-vX.Y.Z.zip: Calico for Windows zip archive for non-HPC installation.
+// - calico-windows-vX.Y.Z.zip: Calico for Windows.
```
Comment (coutinhop) [inline hack/release/pkg/builder/builder.go:291]: I think we need to revert this for v3.27 (which is when HPC support begins)

File: hack/release/pkg/builder/builder.go [hunk 3]
```diff
@@ -445,33 +444,17 @@ func (r *ReleaseBuilder) buildContainerImages(ver string) error {
...
-		out, err := r.makeInDirectoryWithOutput(dir, "release-build", env...)
-		if err != nil {
-			logrus.Error(out)
-			return fmt.Errorf("Failed to build %s: %s", dir, err)
-		}
-		logrus.Info(out)
-	}
-
-	for _, dir := range windowsReleaseDirs {
-		out, err := r.makeInDirectoryWithOutput(dir, "image-windows", env...)
```
Comment (coutinhop) [inline hack/release/pkg/builder/builder.go:469]: I think we need to revert this for v3.27

File: hack/release/pkg/builder/builder.go [hunk 4]
```diff
@@ -533,11 +516,6 @@ func (r *ReleaseBuilder) publishContainerImages(ver string) error {
 		"node",
 	}
 
-	windowsReleaseDirs := []string{
-		"node",
-		"cni-plugin",
-	}
-
```
Comment (coutinhop) [inline hack/release/pkg/builder/builder.go:540]: I think we need to revert this for v3.27

File: hack/release/pkg/builder/builder.go [hunk 5]
```diff
@@ -568,25 +546,6 @@ func (r *ReleaseBuilder) publishContainerImages(ver string) error {
...
-				}
-				logrus.Error(out)
-				return fmt.Errorf("Failed to publish %s: %s", dir, err)
-			}
-
-			// Success - move on to the next directory.
-			logrus.Info(out)
-			break
-		}
-	}
```
Comment (coutinhop) [inline hack/release/pkg/builder/builder.go:589]: I think we need to revert this for v3.27

**Review by coutinhop — COMMENTED** [review]
We'll start supporting HPC and will push the `calico/node-windows` and `calico/cni-windows` images on v3.27, made a few comments on the code I don't think should have been deleted

**Pull request merged**